### PR TITLE
fix(Email Trigger (IMAP) Node): Ensure connection close does not block deactivation

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/PAY-1919/imap-connection-closed-unexpectedly#comment-979ca863

Follow-up to #10064

Related to #8738

### CODE REVIEW

File: packages/nodes-base/nodes/EmailReadImap/v2/EmailReadImapV2.node.ts [hunk 1]
```diff
@@ -669,14 +669,19 @@ export class EmailReadImapV2 implements INodeType {
...
 			closeFunctionWasCalled = true;
 			if (reconnectionInterval) {
 				clearInterval(reconnectionInterval);
 			}
-			if (connection.closeBox) await connection.closeBox(false);
-			connection.end();
-		}
+			try {
+				if (connection.closeBox) await connection.closeBox(false);
+				connection.end();
```
Comment (tomi) [inline packages/nodes-base/nodes/EmailReadImap/v2/EmailReadImapV2.node.ts:679]: Should the `connection.end` be in a `finally` block?

Comment (ivov) [reply packages/nodes-base/nodes/EmailReadImap/v2/EmailReadImapV2.node.ts:679]: This cannot fail? https://github.com/mscdex/node-imap/blob/9918f08bc0bf5e3b0ed0e154fe98b3396e108606/lib/Connection.js#L306

**Review by ShireenMissi — APPROVED** [review]
LGTM 🎉

**Comment by janober** [comment]
Got released with [`n8n@1.58.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.58.1)

**Pull request merged**